qt.lua

```diff
@@ -575,6 +579,58 @@ function premake.extensions.qt.getQRCDependencies(fcfg)
 
 end
 
+--
+-- Checks if a file is a ts file.
+--
+-- @param filename
+--		The file name to check.
+-- @return
+--		true if the file needs to be run through the lrelease tool, false if not.
+--
+function premake.extensions.qt.isTS(filename)
+	return path.hasextension(filename, { ".ts" })
+end
+
+--
+-- Adds the custom build for a ts file.
+--
+-- @param fcfg
+--		The config for a single file.
+-- @param cfg
+--		The config of the project ?
+--
+function premake.extensions.qt.addTSCustomBuildRule(fcfg, cfg)
+	local qt = premake.extensions.qt
+
+	-- get the project's location (to make paths relative to it)
+	local projectloc = fcfg.project.location
+
+	-- create the output file name
+	local qtqmgenerateddir = fcfg.qtqmgenerateddir or cfg.qtqmgenerateddir
+	local output = path.join(qtqmgenerateddir or cfg.targetdir, fcfg.basename .. ".qm")
+
+	-- create the moc command
+	local command = '"' .. fcfg.config.qtbinpath .. '/lrelease" "' .. fcfg.relpath .. '"'
+	command = command .. ' -qm "' .. path.getrelative(projectloc, output) .. '"'
+
+	-- now create the arguments
+	local arguments = ""
+
+	-- if we have custom commands, add them
+	local qtlreleaseargs = iif(#fcfg.qtlreleaseargs ~= 0, fcfg.qtlreleaseargs, cfg.qtlreleaseargs)
```

Comment:
I just noticed that you use this `iif` pattern in this function to lookup the arguments in the file config and then in the config, but is that necessary? I was pretty sure that `fcfc` kinda inherits `cfg` configs... (at least in the rest of the addon, I never have to lookup in the `cfg` object)

Reply:
Hey,

Sorry for the late reply! Well, judging from this, your code and mine are both incorrect (mine only take into account `fcfg`, which I thought already contained `cfg`...) and yours take either `fcfg` or `cfg`, and it'll only work when there wasn't any file filter.

I'll have to double check that for all the other apis, and create a `merge` helper function to merge `cfg` and `fcfg` (at least for the options we're interested in.

Reply:
A `merge` helper would be fine indeed.
Not sure how you want we continue with the PR.
It seems I just have to wait on my side.

Reply:
Hey, sorry for the wait! I've fixed all the custom command arguments, you can update and rebase, and you'll be able to do like it's done with uic, etc.

Reply:
Changed and rebased.